### PR TITLE
Updated circle ci config file to 2.0 by updating wp-cli to 2.0.0

### PR DESCRIPTION
**Pull request by angtheod**
Issue https://github.com/greenpeace/planet4-plugin-blocks/issues/335

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    command: |
+      sudo apt-get update && sudo apt-get install subversion
+      sudo docker-php-ext-install mysqli
+      sudo sh -c "printf '\ndeb http://ftp.us.debian.org/debian sid main\n' >> /etc/apt/sources.list"
+      sudo apt-get update && sudo apt-get install mysql-client-5.7
+
+  php_job: &php_job
+    environment:
+      - WP_TESTS_DIR: "/tmp/wordpress-tests-lib"
+      - WP_CORE_DIR: "/tmp/wordpress/"
```
Comment (angtheod) [inline .circleci/config.yml]: @27Bslash6 is it ok if it downloads the wordpress_tests suite inside /tmp?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    steps:
+      - checkout
+      - run: *setup_environment
+      - run: *install_dependencies
+      - run:
+          name: "Run Tests"
+          command: |
+            composer global require "phpunit/phpunit=5.7.*"
```
Comment (angtheod) [inline .circleci/config.yml]: @27Bslash6 Should I change that to 6.5.* ? Or it does not have an effect either way?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            composer global require wp-coding-standards/wpcs
+            phpcs --config-set installed_paths $HOME/.composer/vendor/wp-coding-standards/wpcs
+            phpcs
+            rm -rf $WP_TESTS_DIR $WP_CORE_DIR
+            bash bin/install-wp-tests.sh wordpress_test root '' 127.0.0.1 latest
+            phpunit
+            WP_MULTISITE=1 phpunit
+
+jobs:
+  php56-build:
```
Comment (angtheod) [inline .circleci/config.yml]: @27Bslash6 should I remove the php5.6 job?

Comment (27Bslash6) [reply .circleci/config.yml]: Yes we'll never need the 5.6 tests

File: .circleci/config.yml (hunk 1 above)
Comment (27Bslash6) [reply .circleci/config.yml]: Yes, no worries. You can manipulate the file system any way you require.

File: .circleci/config.yml (hunk 2 above)
Comment (27Bslash6) [reply .circleci/config.yml]: I don't know whether the version would have any effect.  If this is boilerplate from `wp scaffold`, we can start with 5.7.* and then update to 6.5 in a future commit.

File: .phpcs.xml.dist [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+			<!-- Value: replace the text domain used. -->
+			<property name="text_domain" type="array" value="my-plugin"/>
+		</properties>
+	</rule>
+	<rule ref="WordPress.WhiteSpace.ControlStructureSpacing">
+		<properties>
+			<property name="blank_line_check" value="true"/>
+		</properties>
+	</rule>
+</ruleset>
```
Comment (27Bslash6) [inline .phpcs.xml.dist]: This file fails to load into phpcs: https://circleci.com/gh/greenpeace/planet4-plugin-blocks/14
```
ERROR: Ruleset /home/circleci/project/.phpcs.xml.dist is not valid

Run "phpcs --help" for usage information

Exited with code 3
```
I've tested this install with the default file https://raw.githubusercontent.com/WordPress-Coding-Standards/WordPress-Coding-Standards/develop/phpcs.xml.dist.sample and phpcs successfully parses the config file, but the style / standards may not be what you're used to.

Can you have a look at this file and see if it's more appropriate, or fix the existing so at least the tests run, if not succeed?

**Comment by 27Bslash6** [comment]
Alright this now executes phpcs and phpunit tests, and stores the results in a format that Circle can summarise. 

Next steps:
- #339 solve the phpcs errors
- increase unit test coverage
- automate the deployment of new releases

**Pull request merged**